zanata-client-commands/src/main/java/org/zanata/client/commands/push/RawPushCommand.java

```diff
@@ -75,10 +94,164 @@ public RawPushCommand(PushOptions opts, RestClientFactory clientFactory) {
         client = getClientFactory().getFileResourceClient();
     }
 
+    /**
+     * Extract fileType from input pattern: fileType[extension;extension]
+     * @param typeWithExtension
+     */
+    private String extractFileType(String typeWithExtension) {
+        if (typeWithExtension.contains("[")) {
+            return typeWithExtension.substring(0,
+                    typeWithExtension.indexOf("[")).toUpperCase();
+        } else {
+            return typeWithExtension.toUpperCase();
+        }
+    }
+
+    /**
+     * Extract extensions in square bracket
+     * input pattern: fileType[extension;extension], [extension;extension]
+     */
+    private List<String> extractExtensions(String typeWithExtension) {
+        Pattern p = Pattern.compile("\\[(.*?)\\]");
+        Matcher m = p.matcher(typeWithExtension);
+        if (m.find()) {
+            return Arrays.asList(m.group(1).split(";"));
+        }
+        return Collections.emptyList();
+    }
+
+
+    /**
+     * Validate inputFileType to make sure it's not empty
+     *
+     * @param inputFileType
+     * @param inputExtensions
+     * @param acceptedTypes
+     */
+    private void validateInputFileType(String inputFileType,
+            List<String> inputExtensions, List<DocumentType> acceptedTypes) {
+        if (StringUtils.isNotBlank(inputFileType)) {
+            return;
+        }
+        if (inputExtensions.isEmpty()) {
+            //throw error if inputFileType and inputExtensions is empty
+            throw new RuntimeException(
+                    "Invalid expression for '--file-types' option");
+        } else {
+            //suggest --file-types options for this extension
+            for (DocumentType docType: acceptedTypes) {
+                for (String extension: docType.getSourceExtensions()) {
+                    if (inputExtensions.contains(extension)) {
+                        String msg = "Please use file type '"
+                            + docType
+                            + "' for extension '"
+                            + extension
+                            + "', e.g. --file-types "
+                            + docType;
+                        throw new RuntimeException(msg);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Return list of validated type with list of extensions
+     *
+     * Validate user input file types against server accepted file types
+     *
+     * - Each file type is only input once
+     * - Same file extension cannot be in multiple file type
+     * - Source file extension that is not supported from file type will be ignored.
```

Comment:
- [ ] Just ignored, or generate a warning? We should at least warn the user that we are ignoring something they input (and if we are, this comment should say so).